core/chainio/avs_writer.go

```diff
@@ -70,10 +72,11 @@ func NewAvsWriterFromConfig(baseConfig *config.BaseConfig, ecdsaConfig *config.E
 	}, nil
 }
 
-func (w *AvsWriter) SendAggregatedResponse(batchIdentifierHash [32]byte, batchMerkleRoot [32]byte, senderAddress [20]byte, nonSignerStakesAndSignature servicemanager.IBLSSignatureCheckerNonSignerStakesAndSignature) (*common.Hash, error) {
+func (w *AvsWriter) SendAggregatedResponse(batchIdentifierHash [32]byte, batchMerkleRoot [32]byte, senderAddress [20]byte, nonSignerStakesAndSignature servicemanager.IBLSSignatureCheckerNonSignerStakesAndSignature) (*types.Receipt, error) {
```

Comment:
for what I understand, this function has two parts:

1. One where a simulation of the transaction is made
2. One part with the logic that is going to be retried if the transaction is not included.

I think it is easier to follow the code if we just encapsulate all the logic in 2 in an anonymous function and just call `Retry`. It may be a little more monolithic but it is not really a problem

Reply:
Updated!